src/core/index.rs

```diff
@@ -1,5 +1,5 @@
 use Result;
-use Error;
+use error::*;
```

Comment:
There's also a `ResultExt` trait that gives us the `chain_err` method. 

It's not uncommon to `use error::*` but if you'd rather avoid it I think that's reasonable too, I can just import the bits from `error` I actually need.